go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -871,6 +871,12 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 					if nc.name != node.Name && util.GetNodeZone(&node) != config.Default.Zone && !houtil.IsHybridOverlayNode(&node) {
 						nodeSubnets, err := util.ParseNodeHostSubnetAnnotation(&node, types.DefaultNetworkName)
 						if err != nil {
+							if annotationError, ok := err.(*util.AnnotationNotSetError); ok {
+								if strings.Contains(annotationError.Msg, fmt.Sprintf("node %q has no k8s.ovn.org/node-subnets annotation for network %s", node.Name, types.DefaultNetworkName)) {
```

Comment:
why do you need to string match here? you already know its an annotation not set, so this seems like overkill and not maintainable long term if the message changes.